lib/types.nix

```diff
@@ -475,6 +475,47 @@ rec {
           else res;
     };
 
+    # This type is like `package` but it gets merged based on the
+    # `meta.priority` attribute.
+    packageByPrio =
```

Comment:
How does this interact with `mkForce` and co.? Those should definitely also work.

Actually `meta.priority` is kind of just a mechanism to get around not having a proper override mechanism before. But now that we do with `mkForce` and co., it shouldn't be necessary anymore.

I'd be in favor of ignoring `meta.priority`'s on this level, or even complaining if they're set (for attribute sets), referring people to `mkForce` and co. instead.

Reply:
Yeah `mkForce` works (see the test case). The priority is currently only used when merging, so only if the same package is set multiple time with the same module-level priority.

We could try to remove it, but it would take a bit more effort to re-encode the current meta priorities as module-system priorities.

Reply:
`meta.priority` is about contents, whereas module level priorities are about replacing actual packages. Those are distinct solutions to distinct problems.
Two interesting cases:
- Remove a command by replacing a package. This can only be done by overriding the `.package` with the module system.
- Resolve a conflict between two packages that both want to create some file. Use `meta.priority` to pick the best one. You probably don't want to delete the other files contributed by the lower package.

Reply:
So do I understand correctly that we are OK with the changes as proposed in this PR currently, or do we need any changes still?

Reply:
This morning I realised that the whole priority concept is actually not relevant here. We don't need a new type if we just change the already existing `package` type to use `mergeEqualOption` (I don't see a reason for using `mergeOneOption` there) so that if we assign the same derivation twice, it will not cause a conflict.

The priority concept is only used while building the final environment containing all system packages and is used to resolve conflicts when the same file is present in multiple derivations.